src/alinea/pyratp/runratp.py

```diff
@@ -1,70 +1,101 @@
-# Header
-#
-
-"""
-"""
-
-from alinea.pyratp import pyratp
-import numpy as np
 import os
+import platform
 import shutil
 import tempfile
+
+import numpy as np
+import pandas as pd
+
 import grid
-import platform
+from alinea.pyratp import pyratp
+
+columns = ('VegetationType',
+           'ntime',
+           'day',
+           'hour',
+           'AirTemperature',
+           'VoxelId',
+           'ShadedTemp',
+           'SunlitTemp',
+           'STARDirect',
+           'STARSky',
+           'ShadedPhoto',
+           'SunlitPhoto',
+           'ShadedTranspi',
+           'SunlitTranspi',
+           'ShadedArea',
+           'SunlitArea',
+           'ShadedGs',
+           'SunlitGs',
+           'ShadedAbsorbedPAR',
+           'SunlitAbsorbedPAR',
+           'ShadedAbsorbedNIR',
+           'SunlitAbsorbedNIR')
+
+columns_tree = ('ntime',
+                'day',
+                'hour',
+                'VegetationType',
+                'TotalIrradiation',
+                'AirTemperature',
+                'TreePhotosynthesis',
+                'TreeTranspiration')
+
 
 class runRATP(object):
     """
     """
+
     @staticmethod
     def DoAll(*args):
         ratp = pyratp.ratp
         pyratp.dir_interception.scattering = False
-        ratp.out_time_spatial = np.zeros(pyratp.micrometeo.nbli*pyratp.grid3d.nveg*22*pyratp.grid3d.nent).reshape(pyratp.micrometeo.nbli*pyratp.grid3d.nveg*pyratp.grid3d.nent,22) 
-        ratp.out_time_tree = np.zeros(pyratp.micrometeo.nbli*8*pyratp.grid3d.nent).reshape(pyratp.micrometeo.nbli*pyratp.grid3d.nent ,8)
+        ratp.out_time_spatial = np.zeros((pyratp.micrometeo.nbli * pyratp.grid3d.nveg * pyratp.grid3d.nent, 22))

```

Comment:
Replace 22 by len(columns).
And 8 by len(columns_tree).
Otherelse it is cryptic: Is 8 refers to 8 bytes... it takes me some seconds to figure out what 22 and 8 means.